### PR TITLE
Feature/git actions

### PR DESCRIPTION
**Pull request by milospp**
## Description

Created a [CI pipeline](https://github.com/chenejac/TeslaRIS-backend/issues/2) containing 4 jobs:
- BUILD
- TEST
- ANALYZE
- CODE STYLE

Added simple pull request description template

## Checklist

-   [x] I have tested these changes locally
-   [x] I have added tests that cover these changes (if applicable)
-   [x] I have updated the documentation (if applicable)
-   [x] I have added or updated the necessary comments in the code

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+## Description
+
+[Enter a brief description of the changes included in this pull request]
+
+## Checklist
```
Comment (chenejac) [inline .github/PULL_REQUEST_TEMPLATE.md:5]:
```suggestion
## How should this be tested?

A description of what steps someone could take to:

- Reproduce the problem you are fixing (if applicable)
- Test that the pull request does what is intended.

## Checklist
```

File: checkstyle-maven.xml [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+        <module name="SuppressionXpathSingleFilter">
+            <!-- suppresion is required till https://github.com/checkstyle/checkstyle/issues/7541 -->
+            <property name="id" value="RightCurlyAlone"/>
+            <property name="query" value="//RCURLY[parent::SLIST[count(./*)=1]
+                                     or preceding-sibling::*[last()][self::LCURLY]]"/>
+        </module>
+        <module name="WhitespaceAfter">
+            <property name="tokens"
+                      value="COMMA, SEMI, TYPECAST, LITERAL_IF, LITERAL_ELSE,
+                    LITERAL_WHILE, LITERAL_DO, LITERAL_FOR, DO_WHILE"/>
```
Comment (chenejac) [inline checkstyle-maven.xml:107]:
```suggestion
                                            value="COMMA, SEMI, TYPECAST, LITERAL_IF, LITERAL_ELSE, LITERAL_RETURN,
                    LITERAL_WHILE, LITERAL_DO, LITERAL_FOR, LITERAL_FINALLY, DO_WHILE, ELLIPSIS,
                    LITERAL_SWITCH, LITERAL_SYNCHRONIZED, LITERAL_TRY, LITERAL_CATCH, LAMBDA,
                    LITERAL_YIELD, LITERAL_CASE"/>
```

File: checkstyle-maven.xml [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+                      value="LITERAL_TRY, LITERAL_CATCH, LITERAL_FINALLY, LITERAL_IF, LITERAL_ELSE,
+                    LITERAL_DO"/>
+        </module>
+        <module name="RightCurly">
+            <property name="id" value="RightCurlyAlone"/>
+            <property name="option" value="alone"/>
+            <property name="tokens"
+                      value="CLASS_DEF, METHOD_DEF, CTOR_DEF, LITERAL_FOR, LITERAL_WHILE, STATIC_INIT,
+                    INSTANCE_INIT, ANNOTATION_DEF, ENUM_DEF, INTERFACE_DEF, RECORD_DEF,
+                    COMPACT_CTOR_DEF"/>
```
Comment (chenejac) [inline checkstyle-maven.xml:96]:
```suggestion
                     COMPACT_CTOR_DEF, LITERAL_SWITCH"/>
```

**Review by chenejac — CHANGES_REQUESTED** [review]
@milospp thanks for the contribution. I have added a couple of comments. Don't understand why we need both - checkstyle.xml and checkstyle-maven.xml? I have added in the form of suggestions the difference between those two files. Can you please double check whether it is necessary to have both files? Also, I have noticed that you added build, test, codestyle action into one file called maven.yml, I would rename that file into contribution.yml or something as that. If I understood well it is for PR over main branch and for any push (even on feature branches), right?

**Comment by ivanmrsulja** [comment]
I have made the changes you requested.

**Review by chenejac — APPROVED** [review]
@ivanmrsulja @milospp thanks, the feature completed

**Pull request merged**